plugin-localization/src/main/java/com.mapbox.mapboxsdk.plugins.localization/LocalizationPlugin.java

```diff
@@ -0,0 +1,234 @@
+    this.mapboxMap = mapboxMap;
+    this.selectedBackupLanguage = backupLanguage;
+    setMapTextLanguage(getDeviceLanguage());
+  }
+
+  /**
+   * Retrieve the code for the language that the device is currently set to.
+   *
+   * @return The correct code according to ISO-639-1
+   */
+  public String getDeviceLanguage() {
+    if (Build.VERSION.SDK_INT >= 21) {
+      String languageTag = Locale.getDefault().toLanguageTag();
+      if (languageTag.contains("Hans")) {
+        return "zh-Hans";
+      } else if (languageTag.contains("Hant")) {
+        return "zh";
+      } else {
+        return Locale.getDefault().toLanguageTag().substring(0, 2);
+      }
+    } else {
+      if (Locale.getDefault().getLanguage().equalsIgnoreCase("IW")) {
+        return "he";
+      } else if (Locale.getDefault().getLanguage().equalsIgnoreCase("IN")) {
+        return "id";
+      } else if (Locale.getDefault().getLanguage().equalsIgnoreCase("JI")) {
+        return "yi";
+      } else if (Locale.getDefault().getLanguage().equalsIgnoreCase("JW")) {
+        return "jv";
+      } else if (Locale.getDefault().getLanguage().contains("zh-Hans")
+        || Locale.getDefault().getDisplayName().contains("简体")) {
+        return "zh-Hans";
+      } else if (Locale.getDefault().getLanguage().contains("zh-Hant")
+        || Locale.getDefault().getDisplayName().contains("繁體")) {
+        return "zh";
+      } else {
+        return Locale.getDefault().getLanguage();
+      }
+    }
+  }
+
+  /**
+   * Checks whether a certain language is supported by Mapbox Streets style source.
+   *
+   * @param languageTag the language tag to check against Mapbox-supported languages
+   * @return whether a certain language (tag) matches one of the tags associated with a
+   * OSM-supported language
+   */
+  private boolean mapboxSourceSupportsLanguage(String languageTag) {
+    return languageTag.equals("ar")
+      || languageTag.equals("en")
+      || languageTag.equals("es")
+      || languageTag.equals("fr")
+      || languageTag.equals("de")
+      || languageTag.equals("pt")
+      || languageTag.equals("ru")
+      || languageTag.equals("zh")
+      || languageTag.equals("zh-Hans");
+  }
+
+  /**
+   * Retrieves and adjusts the individual map layers that need to be adjusted
+   *
+   * @param languageToSetMapTo the language that the map is being adjusted to
+   */
+  private void adjustMapTextLayers(String languageToSetMapTo) {
+    for (Source source : mapboxMap.getSources()) {
+      if (sourceIsFromMapbox(source)) {
+        for (Layer layer : mapboxMap.getLayers()) {
+          if (layerHasAdjustableTextField(layer)) {
+            if (((SymbolLayer) layer).getTextField().getValue().contains("{name")
+              || !getDeviceLanguage().equals("en") && ((SymbolLayer) layer).getTextField().getValue().contains("{abbr}")
+              ) {
+              layer.setProperties(textField(String.format("{name_%s}", languageToSetMapTo)));
```

Comment:
If `text-field` was originally set to something like `{name_en} ({name})`, this would replace it with just `{name_en}` for English speakers. I think it would be better to get the text field value as a string, then find and replace `{name}` and `{name_*}` with `{name_xy}` (where _xy_ is `languageToSetMapTo`), then set the modified string as the text field.

Reply:
I addressed this issue by using a regrex and replacing only the {name_*} part in textfields.